From 8920dd223a8e251b82ea52c8307082cb0a47db89 Mon Sep 17 00:00:00 2001
From: Dinh Nguyen <dinguyen@altera.com>
Date: Tue, 23 Apr 2013 14:13:09 -0500
Subject: [PATCH 077/248] FogBugz #113756: Fix error for FPGA LED PIO base
 address

The base address for the LED PIO is incorrect. The current address is for
the LWH2F bridge and 0xff2100040 is the address for the LED PIO slave.
The current implementation works because Qsys selected the LED PIO as the
default slave, so all reads/writes from the LWH2F bridge master targeting
holes in the master address map gets routed to the LED PIO anywa.

However, if any peripheral with an address range larger than the PIO's gets
added to the system, then the build malfunctions, since that peripheral
becomes the new default slave.

[Original patch taken from
git://git.rocketboards.org/linux-socfpga.git socfpga-3.14]

Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
Signed-off-by: Dinh Nguyen <dinguyen@altera.com>
---
 arch/arm/boot/dts/socfpga.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/socfpga.dtsi b/arch/arm/boot/dts/socfpga.dtsi
index abe02d8..76e4108 100644
--- a/arch/arm/boot/dts/socfpga.dtsi
+++ b/arch/arm/boot/dts/socfpga.dtsi
@@ -69,7 +69,7 @@
 		agpio0: gpio@0xc0000000 {
 			compatible = "altr,pio-1.0";
 			/* Register base 0xff200000 is for a light-weight bridge */
-			reg = <0xff200000 0x10>;
+			reg = <0xff210040 0x10>;
 			width = <32>;
 			/* There are 64 interrupts from the FPGA start at 72, so 45 has to be wrong */
 			interrupts = <0 45 4>;
-- 
1.9.1

